src/test/java/com/uid2/admin/vertx/SharingServiceTest.java

```diff
@@ -340,27 +308,21 @@ void listSiteSetConcurrency(Vertx vertx, VertxTestContext testContext) {
                 "    \"hash\": " + keysets.get(3).hashCode() + "\n" +
                 "  }";
 
-        post(vertx, "api/sharing/list/5", body1, ar -> {
-            assertTrue(ar.succeeded());
-            HttpResponse response = ar.result();
+        post(vertx, testContext, "api/sharing/list/5", body1, response -> {
             assertEquals(200, response.statusCode());
 
             AdminKeyset expected = new AdminKeyset(3, 5, "test", Set.of(22, 25, 6), Instant.now().getEpochSecond(), true, true, new HashSet<>());
-            compareKeysetListToResult(expected, response.bodyAsJsonObject().getJsonArray("whitelist"));
+            compareKeysetListToResult(expected, response.bodyAsJsonObject().getJsonArray("allowed_sites"));
```

Comment:
👀 Nice catch. Guessing the test was being marked complete before this could run?

Reply:
Yes, before the second api call is ended first and complete the test